File: apps/api/app/models/treatment.py
"""Treatment model."""

import uuid
from datetime import date, datetime

from sqlalchemy import Date, DateTime, ForeignKey, String, Text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.base import Base


class Treatment(Base):
    __tablename__ = "treatments"

    hive_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("hives.id", ondelete="CASCADE"), nullable=False, index=True
    )
    treatment_type: Mapped[str] = mapped_column(String(100), nullable=False)
    product_name: Mapped[str | None] = mapped_column(String(255), nullable=True)
    method: Mapped[str | None] = mapped_column(String(100), nullable=True)
    started_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), nullable=False
    )
    ended_at: Mapped[datetime | None] = mapped_column(
        DateTime(timezone=True), nullable=True
    )
    dosage: Mapped[str | None] = mapped_column(String(100), nullable=True)
    effectiveness_notes: Mapped[str | None] = mapped_column(Text, nullable=True)
    follow_up_date: Mapped[date | None] = mapped_column(Date, nullable=True)

    # Relationships
    hive: Mapped["Hive"] = relationship("Hive", back_populates="treatments")

File: apps/mobile/database/models/Treatment.ts
import { Model } from "@nozbe/watermelondb";
import { text, field, date, relation, readonly } from "@nozbe/watermelondb/decorators";
import type { Associations } from "@nozbe/watermelondb/Model";
import type { Relation } from "@nozbe/watermelondb/Model";
import type Hive from "./Hive";

export default class Treatment extends Model {
  static table = "treatments";

  static associations: Associations = {
    hives: { type: "belongs_to" as const, key: "hive_id" },
  };

  @text("hive_id") hiveId!: string;
  @text("treatment_type") treatmentType!: string;
  @text("product_name") productName!: string | null;
  @text("method") method!: string | null;
  @field("started_at") startedAt!: number | null;
  @field("ended_at") endedAt!: number | null;
  @text("dosage") dosage!: string | null;
  @text("effectiveness_notes") effectivenessNotes!: string | null;
  @text("follow_up_date") followUpDate!: string | null;
  @readonly @date("created_at") createdAt!: Date;
  @readonly @date("updated_at") updatedAt!: Date;

  @relation("hives", "hive_id") hive!: Relation<Hive>;
}
